### PR TITLE
Bump Siddhi version to 5.0.0

### PR DESCRIPTION
**Pull request by lasanthaS**
## Purpose
This PR bumps the Siddhi version of the extension to 5 and refactor the code.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

**Review by grainier — APPROVED** [review]

File: component/src/main/java/io/siddhi/extension/execution/reorder/AlphaKSlackExtension.java [hunk 1]
```diff
@@ -205,218 +199,202 @@
...
-
-                    if (timerFlag) {
-                        timerFlag = false;
-                        lastScheduledTimestamp = lastScheduledTimestamp + timerDuration;
-                        scheduler.notifyAt(lastScheduledTimestamp);
-                    }
+                           StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater,
+                           AlphaKSlackState state) {
+        synchronized (state) {
+            ComplexEventChunk<StreamEvent> complexEventChunk = new ComplexEventChunk<StreamEvent>(false);
```
Comment (suhothayan) [inline component/src/main/java/io/siddhi/extension/execution/reorder/AlphaKSlackExtension.java]: Can you test with new ComplexEventChunk<StreamEvent>(true); 
if all test passes please make it true

Comment (lasanthaS) [reply component/src/main/java/io/siddhi/extension/execution/reorder/AlphaKSlackExtension.java]: Both `KSlackExtension` and `AlphaKSlackExtension` passes with `ComplexEventChunk(true)`. Fixed this in commit 2e2e476.

File: component/src/main/java/io/siddhi/extension/execution/reorder/AlphaKSlackExtension.java [hunk 2]
```diff
@@ -588,19 +564,82 @@ public Scheduler getScheduler() {
...
     public void setScheduler(Scheduler scheduler) {
         this.scheduler = scheduler;
-        if (lastScheduledTimestamp < 0 && flag) {
-            lastScheduledTimestamp = siddhiAppContext.getTimestampGenerator().currentTime() +
-                    timerDuration;
-            scheduler.notifyAt(lastScheduledTimestamp);
-        }
     }
 
     private double calculateAlpha(double windowCoverageThreshold, double runtimeWindowCoverage) {
```
Comment (suhothayan) [inline component/src/main/java/io/siddhi/extension/execution/reorder/AlphaKSlackExtension.java]: Pass the sate to the calculateAlpha and you don't need to fetch that every time

File: component/src/main/java/io/siddhi/extension/execution/reorder/KSlackExtension.java [hunk 3]
```diff
@@ -106,40 +111,34 @@
...
-//    private int printcount;
+
+    private List<Attribute> attributes = new ArrayList<>();
+    private SiddhiAppContext siddhiAppContext;
 
     @Override
     public void start() {
-        //Do nothing
+        if (lastScheduledTimestamp < 0) {
+            lastScheduledTimestamp = this.siddhiAppContext.getTimestampGenerator().currentTime() + timerDuration;
```
Comment (suhothayan) [inline component/src/main/java/io/siddhi/extension/execution/reorder/KSlackExtension.java]: lastScheduledTimestamp, eventTreeMap, expiredEventTreeMap and other states that get modified need to be moved into the state. 
That's a bug in the current code

Comment (lasanthaS) [reply component/src/main/java/io/siddhi/extension/execution/reorder/KSlackExtension.java]: Fixed the issue in 2951106.

File: component/src/main/java/io/siddhi/extension/execution/reorder/AlphaKSlackExtension.java (hunk 2 above)
Comment (lasanthaS) [reply component/src/main/java/io/siddhi/extension/execution/reorder/AlphaKSlackExtension.java]: Fixed this in 2951106.

**Pull request merged**